lib/mongo/socket/ssl.rb

```diff
@@ -116,30 +118,43 @@ def connectable?
 
       def create_context(options)
         context = OpenSSL::SSL::SSLContext.new
-        set_cert(context, options) if options[:ssl_cert]
```

Comment:
Removed the if branches because anyway this will be checked in the set_xxx methods